sulong/projects/com.oracle.truffle.llvm.runtime/src/com/oracle/truffle/llvm/runtime/LLVMContext.java

```diff
@@ -87,6 +89,15 @@
 import com.oracle.truffle.llvm.runtime.types.Type;
 
 public final class LLVMContext {
+    // my code start
+    // is object ok here?
+    public final ConcurrentMap<Long, Object> condStorage;
```

Comment:
Try to use the least generic class possible. I rather have a problem with the `Long` since that may prevent your code from working in Sulong's sandboxed mode.

Reply:
i used object here because when using the specific types i got a circular reference warning, and wasn't sure if that's ok

Reply:
That's interesting. I'm not sure where that could come from. In general, compiler errors and warnings result in a failure of the `mx gate` command. They should either be resolved or explicitly suppressed if you can argue why the warning is wrong in your context.